### PR TITLE
Bump .NET versions

### PR DESCRIPTION
**Pull request by Kielek**
## Why

To mitigate https://msrc.microsoft.com/update-guide/vulnerability/CVE-2022-30184

## What
Bump .NET versions
3.1.416 => 3.1.420
5.0.405 => 5.0.408
6.0.200 => 6.0.301

## Tests

CI

### CODE REVIEW

**Review by pellared — APPROVED** [review]

**Review by lachmatt — APPROVED** [review]

**Review by pellared — CHANGES_REQUESTED** [review]
I think we should also update

- `alpine.dockerfile`
- `debian.dockerfile`

In `tracer/build/_build/docker` directory

File: profiler/src/ProfilerEngine/Datadog.Profiler.Native/Datadog.Profiler.Native.vcxproj [hunk 1]
```diff
@@ -24,7 +24,7 @@
     <Keyword>Win32Proj</Keyword>
     <ProjectGuid>{0831386b-1a96-443f-b02b-c1d2b65fd845}</ProjectGuid>
     <RootNamespace>DatadogAutoInstrumentationProfilerNative</RootNamespace>
-    <WindowsTargetPlatformVersion>10.0.18362.0</WindowsTargetPlatformVersion>
+    <WindowsTargetPlatformVersion>10.0</WindowsTargetPlatformVersion>
```
Comment (RassK) [inline profiler/src/ProfilerEngine/Datadog.Profiler.Native/Datadog.Profiler.Native.vcxproj:27]: Any good reason losing patch version here?

Comment (Kielek) [reply profiler/src/ProfilerEngine/Datadog.Profiler.Native/Datadog.Profiler.Native.vcxproj:27]: This version is not available on Windows 2022. What is more, for now, we are not using this library,

Comment (pellared) [reply profiler/src/ProfilerEngine/Datadog.Profiler.Native/Datadog.Profiler.Native.vcxproj:27]: Can we use a version which is available? Or remove this line?

Comment (Kielek) [reply profiler/src/ProfilerEngine/Datadog.Profiler.Native/Datadog.Profiler.Native.vcxproj:27]: 10.0 is used in all other files. Only DD-profiler has different configuration. See https://github.com/signalfx/signalfx-dotnet-tracing/blob/5417e4dd6a8a8e37aeaef638242e4217b81f345c/tracer/src/Datadog.Trace.ClrProfiler.Native/Datadog.Trace.ClrProfiler.Native.DLL.vcxproj#L25

**Comment by Kielek** [comment]
> I think we should also update
> 
> * `alpine.dockerfile`
> * `debian.dockerfile`
> 
> In `tracer/build/_build/docker` directory

Fixed.

File: profiler/src/ProfilerEngine/Datadog.Profiler.Native/Datadog.Profiler.Native.vcxproj (hunk 1 above)
Comment (pellared) [reply profiler/src/ProfilerEngine/Datadog.Profiler.Native/Datadog.Profiler.Native.vcxproj:27]: AFAIK we should try to pin the version if possible. We can deal with it in a separate PR.

**Review by pellared — APPROVED** [review]

**Review by pjanotti — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -320,6 +320,26 @@ jobs:
     #     Import-Module "C:/Program Files/Azure Cosmos DB Emulator/PSModules/Microsoft.Azure.CosmosDB.Emulator"
     #     Start-CosmosDbEmulator -Timeout 300
     # Workaround around long name being hit in MultiDomainHostTests.WorksOutsideTheGAC tests
+    - name: install Microsoft.Net.Component.4.6.1.TargetingPack
```
[the remark this answers is not in the document]
Comment (Kielek) [reply .github/workflows/ci.yml:323]: You are right.

**Review by RassK — APPROVED** [review]

File: profiler/src/ProfilerEngine/Datadog.Profiler.Native/Datadog.Profiler.Native.vcxproj (hunk 1 above)
Comment (Kielek) [reply profiler/src/ProfilerEngine/Datadog.Profiler.Native/Datadog.Profiler.Native.vcxproj:27]: Separate task created.

**Pull request merged**